pkg/service/webhook/service.go

```diff
@@ -83,3 +163,56 @@ func (s Service) GetSupportedNouns() GetSupportedNounsResponse {
 func (s Service) GetSupportedVerbs() GetSupportedVerbsResponse {
 	return GetSupportedVerbsResponse{Verbs: []Verb{Create, Delete}}
 }
+
+func (s Service) PublishWebhook(noun Noun, verb Verb, payload interface{}) {
+	webhook, err := s.storage.GetWebhook(context.Background(), string(noun), string(verb))
+	if err != nil {
+		logrus.WithError(err).Warn("get webhook")
+	}
+
+	if webhook == nil {
+		return
+	}
+
+	for _, url := range webhook.URLS {
+		postPayload := Payload{Noun: noun, Verb: verb, URL: url, Data: payload}
+
+		postJSONData, err := json.Marshal(postPayload)
+		if err != nil {
+			logrus.Warn("marshal payload")
+		}
+		_, err = s.post(url, string(postJSONData))
+		if err != nil {
+			logrus.Warnf("posting payload to %s", url)
+		}
+	}
+}
+
+func (s Service) post(url string, json string) (string, error) {
+	req, err := http.NewRequest(http.MethodPost, url, bytes.NewBuffer([]byte(json)))
+	if err != nil {
+		return "", errors.Wrap(err, "building http req")
+	}
+
+	req.Header.Set("Content-Type", "application/json")
+
+	resp, err := s.httpClient.Do(req)
+	if err != nil {
+		return "", errors.Wrap(err, "client http client")
+	}
+
+	body, err := io.ReadAll(resp.Body)
+	if err != nil {
+		return "", errors.Wrap(err, "parsing body")
+	}
+
+	if is2xxResponse(resp.StatusCode) {
+		return "", fmt.Errorf("status code %v not in the 200s. body: %s", resp.StatusCode, string(body))
+	}
+
+	return string(body), err
+}
+
+func is2xxResponse(statusCode int) bool {
```

Comment:
I saw @andresuribe87 make a change around this recently, is it in a common util package we could pull from?